content/blog/devtools_ui.md

```diff
@@ -0,0 +1,108 @@
+---
+title: "Chrome DevTools UI"
+date: 2021-01-11T09:00:00+02:00
+description: "Chrome DevTools has always been an essential feature in Sauce Labs for live testing real devices. This blog post reports about our recent revamp of the UI code."
+author: "Sakhi Mansoor"
+images:
+    - /images/blog/devtools-ui/after.jpg
+tags:
+    - ux
+    - frontend
+    - chrome
+    - devtools
+    - devtools-ui
+---
+
+Chrome DevTools has always been an essential feature in Sauce Labs for live testing real devices. This will allow a user to inspect web pages in Android devices during live testing and use Chrome DevTools to debug issues.
+
+## Background
+
+In 2016, we launched a Chrome DevTools application to our customers that enabled them to debug webpages in Chrome and Safari in Android and iOS devices respectively. We used the static asset provided by the Chromium team and cloned the whole repository of [artifacts](https://www.google.com/url?q=https://github.com/ChromeDevTools/devtools-frontend/tree/master/front_end&sa=D&ust=1610370981658000&usg=AOvVaw0F69KLsKNRZ2R6ENTiGAKs) in our product from their git source to render the DevTools UI. Some other browsers which use the Chromium DevTools protocol were also integrated with some limitations.
+
+{{< figure src="/images/blog/devtools-ui/before.jpg" class="img-responsive text-center" attr="Here is a screehost of a live testing session debugging Safari on iPad." >}}
+
+In late 2019, we started seeing some serious issues in the Chrome DevTools feature, as some browser APIs were deprecated. The Chromium team rolled out those changes in early 2019 as they migrated all of the DevTools UI to web components, which helped in maintaining the huge codebase and improving their developer experience with more generic guidelines. This article DevTools architecture refresh: migrating to Web Components entails more details on it. The problem was bigger than it seemed to be initially.
```

Comment:
`This article` should be link to their migration details article
link: https://developers.google.com/web/updates/2020/12/migrating-to-web-components